File: src/main/java/be/breina/openrgb/model/Led.java
package be.breina.openrgb.model;

import static lombok.AccessLevel.NONE;

import java.awt.Color;
import java.io.IOException;
import java.util.function.Consumer;
import lombok.Getter;
import lombok.ToString;
import be.breina.openrgb.util.LittleEndianInputStream;

@ToString
@Getter
public class Led {

    private final String name;
    private final int value;

    @Getter(NONE)
    private Consumer<Color> colorSetter;

    Led(LittleEndianInputStream input) throws IOException {
        name = input.readAscii();
        value = input.readInt();
    }

    public void setColor(java.awt.Color color) {
        colorSetter.accept(color);
    }

    void colorMutator(Consumer<java.awt.Color> colorSetter) {
        this.colorSetter = colorSetter;
    }
}

File: src/main/java/be/breina/openrgb/model/Led.kt
package be.breina.openrgb.model

import be.breina.openrgb.util.LittleEndianInputStream
import java.awt.Color
import java.util.function.Consumer

@Suppress("MemberVisibilityCanBePrivate")
class Led private constructor(
    val name: String?,
    val value: Int,
) {

    private lateinit var colorSetter: Consumer<Color>

    fun setColor(color: Color) {
        colorSetter.accept(color)
    }

    internal fun colorMutator(colorSetter: Consumer<Color>) {
        this.colorSetter = colorSetter
    }

    internal companion object {
        fun from(input: LittleEndianInputStream): Led {
            val name = input.readAscii()
            val value = input.readInt()

            return Led(name, value)
        }
    }
}